packages/aws-cdk-lib/aws-appsync/lib/eventapi.ts

```diff
@@ -0,0 +1,584 @@
+    if (props.apiName !== undefined && !Token.isUnresolved(props.apiName)) {
+      if (props.apiName.length < 1 || props.apiName.length > 50) {
+        throw new Error(`\`apiName\` must be between 1 and 50 characters, got: ${props.apiName.length} characters.`);
+      }
+
+      const apiNamePattern = /^[A-Za-z0-9_\-\ ]+$/;
+
+      if (!apiNamePattern.test(props.apiName)) {
+        throw new Error(`\`apiName\` must contain only alphanumeric characters, underscores, hyphens, and spaces, got: ${props.apiName}`);
+      }
+    }
+
+    super(scope, id, {
+      physicalName: props.apiName ?? Lazy.string({
+        produce: () =>
+          Names.uniqueResourceName(this, {
+            maxLength: 50,
+            separator: '-',
+          }),
+      }),
+    });
+
+    this.validateOwnerContact(props.ownerContact);
+
+    this.authProviderTypes = this.setupAuthProviderTypes(props.authorizationConfig?.authProviders);
+
+    const authProviders = props.authorizationConfig?.authProviders ?? [{ authorizationType: AuthorizationType.API_KEY }];
+    this.validateAuthorizationProps(authProviders);
+
+    const connectionAuthModeType = props.authorizationConfig?.connectionAuthModeTypes ?? this.authProviderTypes;
+    const defaultPublishAuthModeTypes = props.authorizationConfig?.defaultPublishAuthModeTypes ?? this.authProviderTypes;
+    const defaultSubscribeAuthModeTypes = props.authorizationConfig?.defaultSubscribeAuthModeTypes ?? this.authProviderTypes;
+
+    this.validateAuthorizationConfig(authProviders, connectionAuthModeType);
+    this.validateAuthorizationConfig(authProviders, defaultPublishAuthModeTypes);
+    this.validateAuthorizationConfig(authProviders, defaultSubscribeAuthModeTypes);
+
+    this.defaultPublishModeTypes = defaultPublishAuthModeTypes;
+    this.defaultSubscribeModeTypes = defaultSubscribeAuthModeTypes;
+
+    if (!Token.isUnresolved(props.ownerContact) && props.ownerContact !== undefined && props.ownerContact.length > 256) {
+      throw new Error('You must specify `ownerContact` as a string of 256 characters or less.');
+    }
+
+    this.eventConfig = {
+      authProviders: this.mapAuthorizationProviders(authProviders),
+      connectionAuthModes: this.mapAuthorizationConfig(connectionAuthModeType),
+      defaultPublishAuthModes: this.mapAuthorizationConfig(defaultPublishAuthModeTypes),
+      defaultSubscribeAuthModes: this.mapAuthorizationConfig(defaultSubscribeAuthModeTypes),
+      logConfig: this.setupLogConfig(props.logConfig),
+    };
+
+    this.api = new CfnApi(this, 'Resource', {
+      name: this.physicalName,
+      ownerContact: props.ownerContact,
+      eventConfig: this.eventConfig,
+    });
+
+    this.api.applyRemovalPolicy(RemovalPolicy.DESTROY);
+
+    this.apiId = this.api.attrApiId;
+    this.apiArn = this.api.attrApiArn;
+    this.dns = this.api.attrDns;
+
+    const apiKeyConfigs = authProviders.filter((mode) => mode.authorizationType === AuthorizationType.API_KEY);
```

Comment:
I believe there's nothiing stopping users to specify multiple` API_KEY` in auth providers now. Then after this filter statement, all the duplicate `API_KEY` will remain. the following statement will create many api keys. However, I believe the intended behvaiour if to create only one api key and ignore the duplicate `API_KEY` specified in the auth provider right?

Reply:
cc @onlybakam to clarify.

Correct - users can specify multiple API keys in the auth providers now. The configurations might be different so the loop iteration after should create as many keys as are specified as auth providers.

Reply:
yes. we allow multiple API keys now. The main check here is making sure that if multiple keys have been specified, they all have names specified. This is the only way to identify a key aftewards. Each created key will have a unieque value.